app/controllers/nonsipp/sharesdisposal/sharesdisposal.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp
+
+import play.api.mvc.Call
+import config.Refined.{Max50, Max5000}
+import models.SchemeId.Srn
+import pages.nonsipp.sharesdisposal.SharesDisposalProgress
+import models.UserAnswers
+import utils.FunctionKUtils._
+import viewmodels.models.SectionJourneyStatus
+
+import scala.concurrent.Future
+
+package object sharesdisposal {
+
+  implicit class CallOps(call: Call) {
+
+    val isCyaPage: Boolean = {
+      val pattern =
+        s"^\\/pension-scheme-return\\/[^\\/]+\\/(change|check-answers)-share-disposal\\/(?!.*\\/\\d+\\/\\d+$$).*"
+
+      call.url.matches(pattern)
+    }
+  }
+
+  def saveProgress(
```

Comment:
This is only called from one place: `HowWereSharesDisposedController` whilst in member payments employer contribution an equivalent is called from multiple places. Is this right here?

Reply:
Yeah this will be used for smart navigation in the future - for the time being it's used to set the completed flag for each completed journey